test/e2e/subscription_e2e_test.go

```diff
@@ -1366,6 +1366,175 @@ var _ = Describe("Subscription", func() {
 		require.ElementsMatch(GinkgoT(), []string{"csv-a", "csv-b", "csv-e"}, got)
 	})
 
+	It("with dependencies from different catsrc with priorities", func() {
```

Comment:
I would be interested in ways to simplify this test. A couple ideas:

- Split into multiple, smaller specs that involve choices between two operators in different catalogs. That would open up the possibility of a table-based test that parameterizes catalog namespace, name, and priority.
- Can catalog preferences be tested without defining any operator-to-operator dependencies at all? Or, can a single subscription that omits catalog namespace and name exercise the same behavior?

If those two are possible, now all cases are reduced to two operator versions, two packages, two catalogs, and one subscription.

Reply:
the 4 priority senarios are completely separable. so table testing is possible. or we could separate this test into 4 smaller tests. 

I can't think of a way to test this without operator-to-operator dependencies. besides, as e2e, I actually prefer to have this dependency because that is their primary function.